frontend/controller/serviceworkers/push.js

```diff
@@ -0,0 +1,134 @@
+import { PUBSUB_INSTANCE } from '@controller/instance-keys.js'
+import { makeNotification } from '@model/notifications/nativeNotification.js'
+import sbp from '@sbp/sbp'
+import setupChelonia from '~/frontend/setupChelonia.js'
+import { NOTIFICATION_TYPE, PUBSUB_RECONNECTION_SUCCEEDED, PUSH_SERVER_ACTION_TYPE, REQUEST_TYPE, createMessage } from '~/shared/pubsub.js'
+
+export default (sbp('sbp/selectors/register', {
+  'push/getSubscriptionOptions': (() => {
+    let cachedVapidInformation
+    return () => {
+      if (
+        cachedVapidInformation &&
+        (performance.now() - cachedVapidInformation[0]) < 3600
```

Comment:
Can you add a comment explaining why 3.6 seconds was chosen for the amount of time to cache this info? (Seems short?)

Reply:
The intent was 3600e3

Reply:
Although, a short period potentially works too (but I'd have made it something like 1 minute in that case). I think the main reason this could be an issue is if trying to set up the subscription many times at once (say, re-opening a browser with many GI tabs open)